sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosSyncStoredProcTest.java

```diff
@@ -148,19 +150,59 @@ public void deleteStoredProcedure() throws Exception {
+            "function() {" +
+                "        var mytext = \"x\";" +
+                "        var myval = 1;" +
+                "        try {" +
+                "            console.log(\"The value of %s is %s.\", mytext, myval);" +
+                "            getContext().getResponse().setBody(\"Success!\");" +
+                "        }" +
+                "        catch(err) {" +
+                "            getContext().getResponse().setBody(\"inline err: [\" + err.number + \"] \" + err);" +
+                "        }" +
+                "}");
+
+        container.getScripts().createStoredProcedure(storedProcedure);
         CosmosStoredProcedureRequestOptions options = new CosmosStoredProcedureRequestOptions();
         options.setPartitionKey(PartitionKey.NONE);
         CosmosStoredProcedureResponse executeResponse = container.getScripts()
-                                                                 .getStoredProcedure(sproc.getId())
+                                                                 .getStoredProcedure(storedProcedure.getId())
                                                                  .execute(null, options);
 
         assertThat(executeResponse.getActivityId()).isNotEmpty();
+        assertThat(executeResponse.getScriptLog()).isNull();
+    }
+
+    @Test(groups = "simple", timeOut = TIMEOUT)
+    public void executeStoredProcedureWithScriptLoggingEnabled() throws Exception {
+        // Create a stored procedure
+        CosmosStoredProcedureProperties storedProcedure = new CosmosStoredProcedureProperties(
+            UUID.randomUUID().toString(),
+            "function() {" +
+                "        var mytext = \"x\";" +
+                "        var myval = 1;" +
+                "        try {" +
+                "            console.log(\"The value of %s is %s.\", mytext, myval);" +
+                "            getContext().getResponse().setBody(\"Success!\");" +
+                "        }" +
+                "        catch(err) {" +
+                "            getContext().getResponse().setBody(\"inline err: [\" + err.number + \"] \" + err);" +
+                "        }" +
+                "}");
+
+        container.getScripts().createStoredProcedure(storedProcedure);
+        CosmosStoredProcedureRequestOptions options = new CosmosStoredProcedureRequestOptions();
+        options.setScriptLoggingEnabled(true);
+        options.setPartitionKey(PartitionKey.NONE);
+
+        CosmosStoredProcedureResponse executeResponse = container.getScripts()
+                                                                 .getStoredProcedure(storedProcedure.getId())
+                                                                 .execute(null, options);
+
+        String logResult = "The value of x is 1.";
+        assertThat(URLDecoder.decode(executeResponse.getScriptLog(), StandardCharsets.UTF_8)).isEqualTo(logResult);
```

Comment:
Nice.

Reply:
I wonder whose job it is to do the decoding? is it the SDK job to decode scriptLog or the user?